test/api_integration/apis/saved_objects/find.js

```diff
@@ -109,6 +109,62 @@ export default function ({ getService }) {
             })
         ));
       });
+
+      describe('with a filter', () => {
+        it('should return 200 with a valid response', async () => (
+          await supertest
+            .get('/api/saved_objects/_find?type=visualization&filter=visualization.attributes.title:"Count of requests"')
+            .expect(200)
+            .then(resp => {
+              expect(resp.body).to.eql({
+                page: 1,
+                per_page: 20,
+                total: 1,
+                saved_objects: [
+                  {
+                    type: 'visualization',
+                    id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+                    attributes: {
+                      title: 'Count of requests',
+                      visState: resp.body.saved_objects[0].attributes.visState,
+                      uiStateJSON: '{"spy":{"mode":{"name":null,"fill":false}}}',
+                      description: '',
+                      version: 1,
+                      kibanaSavedObjectMeta: {
+                        searchSourceJSON: resp.body.saved_objects[0].attributes.kibanaSavedObjectMeta.searchSourceJSON,
+                      },
+                    },
+                    references: [
+                      {
+                        name: 'kibanaSavedObjectMeta.searchSourceJSON.index',
+                        type: 'index-pattern',
+                        id: '91200a00-9efd-11e7-acb3-3dab96693fab',
+                      }
+                    ],
+                    migrationVersion: {
+                      visualization: '7.3.1',
+                    },
+                    updated_at: '2017-09-21T18:51:23.794Z',
+                    version: 'WzIsMV0=',
+                  },
+                ],
+              });
+            })
+        ));
+
+        it('should return 500 with Internal Server Error', async () => (
```

Comment:
Do we want this to be a `400` Bad Request?

Reply:
I re-did like in the find function, throwing a `TypeError` when something is not valid. That's why you get a 500 with Internal Server Error.

Reply:
We always return a 400 for the other errors because the route validation detects the missing parameters and throws a 400 before even calling the SavedObjectsRepository.find function.

It would be safe to catch any `TypeErrors` in the `/api/saved_objects/_find` route handler and return an 400 error with the error.message.

Reply:
Ok talked to @rudolf, now I do understand what both of you are talking about. Let me add a commit to this PR and change the text to match that.
Thank @rudolf and @kobelb for your patience!

Reply:
@kobelb and @rudolf I am using `SavedObjectsErrorHelpers` to throw a 400 Bad Request. It seems to be reasonable for me to use the same technique as other functions do.